Sources/xcproj/PBXProj.swift

```diff
@@ -20,36 +145,28 @@ public class PBXProj: Decodable {
 
     var referenceCache: [String: PBXObject] = [:]
 
-    public var buildFiles: [PBXBuildFile] = []
-    public var aggregateTargets: [PBXAggregateTarget] = []
-    public var containerItemProxies: [PBXContainerItemProxy] = []
-    public var copyFilesBuildPhases: [PBXCopyFilesBuildPhase] = []
-    public var groups: [PBXGroup] = []
-    public var fileElements: [PBXFileElement] = []
-    public var configurationLists: [XCConfigurationList] = []
-    public var versionGroups: [XCVersionGroup] = []
-    public var buildConfigurations: [XCBuildConfiguration] = []
-    public var variantGroups: [PBXVariantGroup] = []
-    public var targetDependencies: [PBXTargetDependency] = []
-    public var sourcesBuildPhases: [PBXSourcesBuildPhase] = []
-    public var shellScriptBuildPhases: [PBXShellScriptBuildPhase] = []
-    public var resourcesBuildPhases: [PBXResourcesBuildPhase] = []
-    public var frameworksBuildPhases: [PBXFrameworksBuildPhase] = []
-    public var headersBuildPhases: [PBXHeadersBuildPhase] = []
-    public var nativeTargets: [PBXNativeTarget] = []
-    public var fileReferences: [PBXFileReference] = []
-    public var projects: [PBXProject] = []
-    public var referenceProxies: [PBXReferenceProxy] = []
-    public var buildPhases: [PBXBuildPhase] {
-        var phases: [PBXBuildPhase] = []
-        phases.append(contentsOf: self.copyFilesBuildPhases as [PBXBuildPhase])
-        phases.append(contentsOf: self.sourcesBuildPhases as [PBXBuildPhase])
-        phases.append(contentsOf: self.shellScriptBuildPhases as [PBXBuildPhase])
-        phases.append(contentsOf: self.resourcesBuildPhases as [PBXBuildPhase])
-        phases.append(contentsOf: self.frameworksBuildPhases as [PBXBuildPhase])
-        phases.append(contentsOf: self.headersBuildPhases as [PBXBuildPhase])
-        return phases
-    }
+    @available(*, deprecated, message: "Use objects.buildFiles instead") public var buildFiles: [PBXBuildFile] { get { return objects.buildFiles.referenceValues } set(v) { objects.buildFiles = Dictionary.from(references: v) } }
```

Comment:
We could move these into a `Deprecations.swift` file that has an extension on PBXProj